filters/log/log.go

```diff
@@ -157,3 +165,52 @@ func (al *auditLog) Response(ctx filters.FilterContext) {
 		log.Errorf("Failed to json encode auditDoc: %v", err)
 	}
 }
+
+type unverifiedAuditLog struct {
+	writer io.Writer
+}
+
+// NewUnverifiedAuditLog logs "Sub" of the middle part of a JWT Token.
+func NewUnverifiedAuditLog() filters.Spec { return &unverifiedAuditLog{writer: os.Stderr} }
+
+func (ual *unverifiedAuditLog) Name() string { return UnverifiedAuditLogName }
+
+// CreateFilter has no arguments. It creates the filter if the user
+// specifies unverifiedAuditLog() in their route.
+func (ual *unverifiedAuditLog) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 0 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	return ual, nil
+}
+
+type jwtMidPart struct {
+	Sub string `json:"sub"`
+}
+
+func (ual *unverifiedAuditLog) Request(ctx filters.FilterContext) {
+	req := ctx.Request()
+	ahead := req.Header.Get(authHeaderName)
+	if !strings.HasPrefix(ahead, authHeaderPrefix) {
+		return
+	}
+
+	fields := strings.FieldsFunc(ahead, func(r rune) bool {
+		return r == []rune(".")[0]
+	})
+	if len(fields) == 3 {
+		sDec, err := base64.RawStdEncoding.DecodeString(fields[1])
+		if err != nil {
+			return
+		}
+
+		var j jwtMidPart
+		err = json.Unmarshal(sDec, &j)
+		if err != nil {
+			return
+		}
+		ual.writer.Write([]byte("Audit sub: " + j.Sub + " "))
```

Comment:
not sure how this works without the newline. I think we need a new line at the end, because even if we rely on the some other following request related log output, there's no guarantee that from the many concurrent requests not something else will print something first.